src/main/java/graphql/schema/PropertyDataFetcher.java

```diff
@@ -3,74 +3,52 @@
 
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
-import java.lang.reflect.Field;
-import java.util.Map;
 
+import static graphql.Assert.assertNotNull;
 import static graphql.Scalars.GraphQLBoolean;
 
 public class PropertyDataFetcher implements DataFetcher {
 
     private final String propertyName;
 
     public PropertyDataFetcher(String propertyName) {
+        assertNotNull(propertyName, "`name` can't be null");
         this.propertyName = propertyName;
     }
 
     @Override
-    public Object get(DataFetchingEnvironment environment) {
-        Object source = environment.getSource();
+    public Object get(DataFetchingEnvironment env) {
+        Object source = env.getSource();
```

Comment:
PropertyDataFetcher implies to me that it can fetch via a Method or via the field directly. If your changing this to only support methods then I'd rename it to MethodDataFetcher.

Personally I believe PropertyDataFetcher being capable of fetching from both a method or a field is much more inline with the Generic Bean Construct and more intuitive than splitting this into two different types of data fetchers.